tests/modules/restful_test/src/Plugin/resource/entity_test/EntityTests__1_0.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful_test\Plugin\resource\entity_test\EntityTests__1_0.
+ */
+
+namespace Drupal\restful_test\Plugin\resource\entity_test;
+
+use Drupal\restful\Plugin\resource\ResourceEntity;
+use Drupal\restful\Plugin\resource\ResourceInterface;
+
+/**
+ * Class EntityTests__1_0
+ * @package Drupal\restful_test\Plugin\resource
+ *
+ * @Resource(
+ *   name = "entity_tests:1.0",
+ *   resource = "entity_tests",
+ *   label = "Entity tests",
+ *   description = "Export the entity test with multiple bundles.",
+ *   authenticationOptional = TRUE,
+ *   dataProvider = {
+ *     "entityType": "entity_test",
+ *     "bundles": {
+ *       "test",
+ *       "main"
+ *     },
+ *   },
+ *   majorVersion = 1,
+ *   minorVersion = 0
+ * )
+ */
+class EntityTests__1_0 extends ResourceEntity implements ResourceInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function publicFields() {
+    $public_fields = parent::publicFields();
+
+    $public_fields['main_bundle'] = array(
```

Comment:
> It requires the client to know about the bundles

Not sure I understand why. We could've called instead of `'main_bundle'` :arrow_right: `'batman!'`, …

> What if when a resource has multiple bundles, you silently skip over non existing fields (only for GET)?

I think that there are two big problems with that approach:
  - This behaviour represents a drupalism where we bundle together things that don't belong together. Outside of Drupal it's hard to understand that _page_ and _article_ are the same thing (node) but a custom entity type _post_ is totally different. So grouping them together beyond the basic common properties is weird.
  - Silently skipping fields altogether goes against [this](https://github.com/RESTful-Drupal/restful/pull/712/files#diff-243b408b86b557a6039177fd9849e028R4). And makes things like automated validation via Apiary or Swagger proxies close to impossible.

Thats why I was thinking that a multiple bundles resource should be a data provider that retrieves data about several bundles, and contains references to the resources that contain the presentation for those individual bundles.

I'd love to hear your thoughts on these points.